ansible/roles/packages/tasks/install-redhat.yaml

```diff
@@ -33,6 +28,25 @@
     - reload systemd
     - enable auditd
 
+# yum-plugin-versionlock is deprecated in rhel 8 and replaced with python3-dnf-plugin-versionlock.
+# python3-dnf-plugin-versionlock has hard dependency on dnf and yum versions which are protected packages on RedHat 8.x and not allowed to upgraded by the OS
+- name: install yum utilities
+  yum:
+    name:
+     - yum-utils
+     - yum-plugin-versionlock
+    state: present
+    enablerepo: "{{ 'offline' if offline_mode_enabled else '' }}"
+    disablerepo: "{{ '*' if offline_mode_enabled else '' }}"
+  register: result
+  until: result is success
+  retries: 5
+  delay: 3
+  notify:
+    - reload systemd
+  when:
+    - ansible_distribution_major_version == '7'
```

Comment:
yum vesionlock plugin has hard dependency on core RHEL protected packages like dnf, yum and incompatible existing packages like dnf-data, python packages etc. this plugin can not be installed without upgrading core packages in "offline" mode. 
NOTE: As a result, we cant lock version of containerd, kubectl etc packages in RHEL 8.x